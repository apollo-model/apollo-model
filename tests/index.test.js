const { query, mutate, CONNECTION, DB } = require('./apolloTest');

import QueryCategories from './queries/queryCategories.graphql';
import CreateCategory from './queries/createCategory.graphql';
import CreateChildCategory from './queries/createChildCategory.graphql';
import QueryCategoriesExtRelation from './queries/queryCategoriesExtRelation.graphql';
import CategoriesFilterByTitle from './queries/categoriesFilterByTitle.graphql';
import CategoriesComplexFilterOr from './queries/categoriesComplexFilterOr.graphql';
import CategoriesRelationFilter from './queries/categoriesRelationFilter.graphql';
import CreateSubscriberWithEmbeddedDocument from './queries/createSubscriberWithEmbeddedDocument.graphql';
import CreateAdmin from './queries/createAdmin.graphql';
import CreatePostWithInterfaceRelation from './queries/createPostWithInterfaceRelation.graphql';
import QueryUsersInterface from './queries/queryUsersInterface.graphql';
import QueryPostsNearPoint from './queries/queryPostsNearPoint.graphql';
import CategoryDelete from './queries/categoryDelete.graphql';
import CategoriesSameFieldFilter from './queries/categoriesSameFieldFilter.graphql';
import CategoriesOrderByFieldWithDBDirective from './queries/categoriesOrderByFieldWithDBDirective.graphql';

jest.setTimeout(10000);

test('QueryCategories empty', async () => {
  let { data } = await query({
    query: QueryCategories,
    variables: {},
  });
  expect(data).toMatchSnapshot();
});

test('CreateCategory', async () => {
  let { data } = await mutate({
    mutation: CreateCategory,
    variables: { title: 'root' },
  });
  expect(data).toMatchSnapshot();
});

test('QueryCategories after create', async () => {
  let { data } = await query({
    query: QueryCategories,
    variables: {},
  });
  expect(data).toMatchSnapshot();
});

test('CreateChildCategory "JS"', async () => {
  let { data } = await mutate({
    mutation: CreateChildCategory,
    variables: { title: 'JS', parentTitle: 'root' },
  });
  expect(data).toMatchSnapshot();
});

test('CreateChildCategory "MongoDB"', async () => {
  let { data } = await mutate({
    mutation: CreateChildCategory,
    variables: { title: 'MongoDB', parentTitle: 'root' },
  });
  expect(data).toMatchSnapshot();
});

test('CreateChildCategory "React"', async () => {
  let { data } = await mutate({
    mutation: CreateChildCategory,
    variables: { title: 'React', parentTitle: 'JS' },
  });
  expect(data).toMatchSnapshot();
});

test('QueryCategoriesExtRelation', async () => {
  let { data } = await query({
    query: QueryCategoriesExtRelation,
    variables: {},
  });
  expect(data).toMatchSnapshot();
});

test('QueryCategoriesByTitle', async () => {
  let { data } = await query({
    query: CategoriesFilterByTitle,
    variables: { title: 'root' },
  });
  expect(data).toMatchSnapshot();
});

test('CategoriesComplexFilterOr', async () => {
  let { data } = await query({
    query: CategoriesComplexFilterOr,
    variables: {},
  });
  expect(data).toMatchSnapshot();
});

test('CategoriesRelationFilter', async () => {
  let { data } = await query({
    query: CategoriesRelationFilter,
    variables: { title: 'root' },
  });
  expect(data).toMatchSnapshot();
});

test('CreateSubscriberWithEmbeddedDocument', async () => {
  let { data } = await query({
    query: CreateSubscriberWithEmbeddedDocument,
    variables: {},
  });
  expect(data).toMatchSnapshot();
});

test('CreateAdmin', async () => {
  let { data } = await query({
    query: CreateAdmin,
    variables: {},
  });
  expect(data).toMatchSnapshot();
});

test('CreatePostWithInterfaceRelation', async () => {
  let { data } = await query({
    query: CreatePostWithInterfaceRelation,
    variables: {},
  });
  expect(data).toMatchSnapshot();
});

test('QueryUsersInterface', async () => {
  let { data } = await query({
    query: QueryUsersInterface,
    variables: {},
  });
  expect(data).toMatchSnapshot();
});

test('QueryPostsNearPoint', async () => {
  let { data } = await query({
    query: QueryPostsNearPoint,
    variables: {},
  });
  expect(data).toMatchSnapshot();
});

test('CategoriesSameFieldFilter', async () => {
  let { data } = await query({
    query: CategoriesSameFieldFilter,
    variables: { title: 'React' },
  });
  expect(data).toMatchSnapshot();
});

test('CategoriesOrderByFieldWithDBDirective', async () => {
  let { data } = await query({
    query: CategoriesOrderByFieldWithDBDirective,
    variables: {},
  });
  expect(data).toMatchSnapshot();
});

test('CategoryDelete', async () => {
  let { data } = await query({
    query: CategoryDelete,
    variables: { title: 'React' },
  });
  expect(data).toMatchSnapshot();
});

test('QueryCategories after delete', async () => {
  let { data } = await query({
    query: QueryCategories,
    variables: {},
  });
  expect(data).toMatchSnapshot();
});

beforeAll(async () => {
  let db = await DB;
  let collections = await db.listCollections().toArray();
  await Promise.all(
    collections.map(({ name }) => db.collection(name).deleteMany({}))
  );
});

afterAll(async () => {
  // Closing the DB connection allows Jest to exit successfully.
  await (await CONNECTION).close();
});
