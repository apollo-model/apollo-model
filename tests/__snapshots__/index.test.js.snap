// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesComplexFilterOr 1`] = `
Object {
  "categories": Array [
    Object {
      "title": "root",
    },
    Object {
      "title": "JS",
    },
  ],
}
`;

exports[`CategoriesRelationFilter 1`] = `
Object {
  "categories": Array [
    Object {
      "title": "JS",
    },
    Object {
      "title": "MongoDB",
    },
  ],
}
`;

exports[`CreateAdmin 1`] = `
Object {
  "createAdmin": Object {
    "username": "admin",
  },
}
`;

exports[`CreateCategory 1`] = `
Object {
  "createCategory": Object {
    "title": "root",
  },
}
`;

exports[`CreateChildCategory "JS" 1`] = `
Object {
  "createCategory": Object {
    "parentCategory": Object {
      "title": "root",
    },
    "title": "JS",
  },
}
`;

exports[`CreateChildCategory "MongoDB" 1`] = `
Object {
  "createCategory": Object {
    "parentCategory": Object {
      "title": "root",
    },
    "title": "MongoDB",
  },
}
`;

exports[`CreateChildCategory "React" 1`] = `
Object {
  "createCategory": Object {
    "parentCategory": Object {
      "title": "JS",
    },
    "title": "React",
  },
}
`;

exports[`CreatePostWithInterfaceRelation 1`] = `
Object {
  "createPost": Object {
    "title": "Build GraphQL API with Apollo",
  },
}
`;

exports[`CreateSubscriberWithEmbeddedDocument 1`] = `
Object {
  "createSubscriber": Object {
    "profile": Object {
      "firstName": "Gwion",
      "lastName": "Britt",
    },
    "username": "subscriber1",
  },
}
`;

exports[`QueryCategories after create 1`] = `
Object {
  "categories": Array [
    Object {
      "title": "root",
    },
  ],
}
`;

exports[`QueryCategories after delete 1`] = `
Object {
  "categories": Array [
    Object {
      "title": "root",
    },
    Object {
      "title": "JS",
    },
    Object {
      "title": "MongoDB",
    },
  ],
}
`;

exports[`QueryCategories empty 1`] = `
Object {
  "categories": Array [],
}
`;

exports[`QueryCategoriesByTitle 1`] = `
Object {
  "categories": Array [
    Object {
      "title": "root",
    },
  ],
}
`;

exports[`QueryCategoriesExtRelation 1`] = `
Object {
  "categories": Array [
    Object {
      "subcategories": Array [
        Object {
          "subcategories": Array [
            Object {
              "title": "React",
            },
          ],
          "title": "JS",
        },
        Object {
          "subcategories": Array [],
          "title": "MongoDB",
        },
      ],
      "title": "root",
    },
    Object {
      "subcategories": Array [
        Object {
          "subcategories": Array [],
          "title": "React",
        },
      ],
      "title": "JS",
    },
    Object {
      "subcategories": Array [],
      "title": "MongoDB",
    },
    Object {
      "subcategories": Array [],
      "title": "React",
    },
  ],
}
`;

exports[`QueryPostsNearPoint 1`] = `
Object {
  "posts": Array [
    Object {
      "place": Object {
        "distance": 1111.9492664453662,
      },
      "title": "Build GraphQL API with Apollo",
    },
  ],
}
`;

exports[`QueryPostsNearPoint 2`] = `
Object {
  "deleteCategory": Object {
    "title": "React",
  },
}
`;

exports[`QueryUsersInterface 1`] = `
Object {
  "users": Array [
    Object {
      "profile": Object {
        "firstName": "Gwion",
        "lastName": "Britt",
      },
      "username": "subscriber1",
    },
    Object {
      "role": null,
      "username": "admin",
    },
    Object {
      "role": null,
      "username": "moderator",
    },
  ],
}
`;
